### PR TITLE
Added HA add-on support the proper way

### PR DESCRIPTION
**Pull request by helto4real**
## Proposed change

Makes config of add-on containers the proper way using environment vars. Thanks @ludeeus for the input. This also allowed us to remove the special host created for add-on.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code compiles without warnings (code quality chek)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/net-daemon/docs

### CODE REVIEW

**Review by FrankBakkerNl — APPROVED** [review]

File: Docker/rootfs/etc/services.d/NetDaemonAddOnApp/run [hunk 1]
```diff
@@ -1,5 +1,16 @@
-#!/usr/bin/with-contenv bash
-echo "Starting NetDaemon Runner"
+#!/usr/bin/with-contenv /usr/bin/bashio
+echo "Starting NetDaemon Runner for add-on"
```
Comment (ludeeus) [inline Docker/rootfs/etc/services.d/NetDaemonAddOnApp/run:2]:
```suggestion

bashio::log.info "Starting NetDaemon Runner for add-on"
```

Comment (ludeeus) [reply Docker/rootfs/etc/services.d/NetDaemonAddOnApp/run:2]: `bashio::log.*` should be used for all echo stements

File: Docker/rootfs/etc/services.d/NetDaemonAddOnApp/run [hunk 2]
```diff
@@ -14,61 +25,58 @@ if [ ! -f "/data/options.json" ]; then
...
+    if [[ "${Netdaemon__ApplicationAssembly}" != /* ]];
     then
-        export NETDAEMON__APPLICATION_ASSEMBLY="/config/netdaemon3/${NETDAEMON__APPLICATION_ASSEMBLY}"
+        export Netdaemon__ApplicationAssembly="/config/netdaemon3/${Netdaemon__ApplicationAssembly}"
     fi 
 
     # The provided application source is ether a project or pre-compiled .Net application
-    if [ ! -f ${NETDAEMON__APPLICATION_ASSEMBLY} ];
+    if [ ! -f ${Netdaemon__ApplicationAssembly} ];
     then
```
Comment (ludeeus) [inline Docker/rootfs/etc/services.d/NetDaemonAddOnApp/run:59]:
```suggestion
    if ! bashio::fs.file_exists "${Netdaemon__ApplicationAssembly}"; then
```

File: Docker/rootfs/etc/services.d/NetDaemonAddOnApp/run [hunk 3]
```diff
@@ -14,61 +25,58 @@ if [ ! -f "/data/options.json" ]; then
...
         
-        echo -e "\\033[31mThe assembly ${NETDAEMON__APPLICATION_ASSEMBLY} cannot be found. Please check the settings.\\033[0m" >&2
+        echo -e "\\033[31mThe assembly ${Netdaemon__ApplicationAssembly} cannot be found. Please check the settings.\\033[0m" >&2
         exit 1
```
Comment (ludeeus) [inline Docker/rootfs/etc/services.d/NetDaemonAddOnApp/run:62]:
```suggestion
        bashio::exit.nok "The assembly ${Netdaemon__ApplicationAssembly} cannot be found. Please check the settings."
```

File: Docker/rootfs/etc/services.d/NetDaemonAddOnApp/run [hunk 4]
```diff
@@ -14,61 +25,58 @@ if [ ! -f "/data/options.json" ]; then
...
     # This is a pre-built deamon
-    echo -e "\\033[32mRunning the pre-built NetDaemon at ${NETDAEMON__APPLICATION_ASSEMBLY}...\\033[0m" >&2
-    cd "$(dirname "${NETDAEMON__APPLICATION_ASSEMBLY}")" || echo -e "\\033[31mCould not change directory to run project\\033[0m" >&2
+    echo -e "\\033[32mRunning the pre-built NetDaemon at ${Netdaemon__ApplicationAssembly}...\\033[0m" >&2
+    cd "$(dirname "${Netdaemon__ApplicationAssembly}")" || echo -e "\\033[31mCould not change directory to run project\\033[0m" >&2
 
-    if [[ "${PWD}" != "$(dirname "${NETDAEMON__APPLICATION_ASSEMBLY}")" ]]; then
+    if [[ "${PWD}" != "$(dirname "${Netdaemon__ApplicationAssembly}")" ]]; then
         echo -e "\\033[31mCould not change directory to run custom project\\033[0m" >&2
         exit 1
```
Comment (ludeeus) [inline Docker/rootfs/etc/services.d/NetDaemonAddOnApp/run:77]:
```suggestion
        bashio::exit.nok "Could not change directory to run custom project"
```

**Pull request merged**